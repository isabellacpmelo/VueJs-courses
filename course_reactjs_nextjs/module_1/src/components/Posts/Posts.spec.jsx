/* eslint-disable testing-library/no-node-access */
import { render, screen } from '@testing-library/react'
import { Posts } from '.'

const props = {
    posts: [
        {
            id: 1,
            title: 'Title Mock 1',
            body: 'Body Mock 1',
            cover: 'img/img1.png'
        },
        {
            id: 2,
            title: 'Title Mock 2',
            body: 'Body Mock 2',
            cover: 'img/img2.png'
        },
        {
            id: 3,
            title: 'Title Mock 3',
            body: 'Body Mock 3',
            cover: 'img/img3.png'
        },
    ],
}

describe('<Posts />', () => {
    test('should render Posts', () => {
        render(<Posts {...props} />)

        expect(screen.getAllByRole('heading', { name: /title/i }))
            .toHaveLength(3)

        expect(screen.getAllByRole('img', { name: /title/i }))
            .toHaveLength(3)

        expect(screen.getAllByText(/body/i))
            .toHaveLength(3)

        expect(screen.getByRole('img', { name: /title mock 3/i }))
        .toHaveAttribute('src', 'img/img3.png')
    })

    test('should match snapshot', () => {
        const {container} = render(<Posts {...props}/>)

        expect(container.firstChild).toMatchSnapshot()
    })
})