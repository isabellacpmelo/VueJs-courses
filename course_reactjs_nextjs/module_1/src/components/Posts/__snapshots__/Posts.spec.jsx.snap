// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title Mock 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title Mock 1
      </h2>
      <p>
        Body Mock 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Mock 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title Mock 2
      </h2>
      <p>
        Body Mock 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Mock 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title Mock 3
      </h2>
      <p>
        Body Mock 3
      </p>
    </div>
  </div>
</div>
`;
